### PR TITLE
Implement attachable tripwire mechanism and tags

### PR DESCRIPTION
**Pull request by SXRWahrheit**
Added tripwire and tripwirehook support to attachable as requested by ULSTICK on Discord: https://discord.com/channels/315163488085475337/1086271537872633937/1086271537872633937

Written with help from @tal5 and tested to be working successfully

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java [hunk 1]
```diff
@@ -8,9 +8,13 @@
 import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.properties.Property;
 import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import net.citizensnpcs.util.NMS;
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java]: Got a few unused imports here

Comment (SXRWahrheit) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java]: Resolved

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java [hunk 2]
```diff
@@ -53,6 +55,7 @@ public static void register() {
         // For a lantern, this returns whether it is hanging from the ceiling.
         // For a gate, this returns whether it is lowered to attach to a wall block.
         // For a mangrove_propagule, this returns whether it is hanging from the block above it.
+        // For a tripwire, denotes whether a tripwire hook or string forms a complete tripwire circuit and is ready to trigger.
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java]: Should follow the format the rest of the entries above use, e.g. `this returns whether...`

**Comment by mcmonkey4eva** [comment]
Do not manually click "Resolve", that is a button for me/aya to use not a PR author

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java [hunk 3]
```diff
@@ -53,6 +55,7 @@ public static void register() {
         // For a lantern, this returns whether it is hanging from the ceiling.
         // For a gate, this returns whether it is lowered to attach to a wall block.
         // For a mangrove_propagule, this returns whether it is hanging from the block above it.
+        // For a tripwire, this returns whether a tripwire hook or string forms a complete tripwire circuit and is ready to trigger.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java:58]: You've also implicitly added support for hanging signs, the behavior of which should be documented

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java [hunk 4]
```diff
@@ -129,6 +147,10 @@ public void adjust(Mechanism mechanism) {
         // For a lantern, this sets whether it is hanging from the ceiling.
         // For a gate, this sets whether it is lowered to attach to a wall block.
         // For a mangrove_propagule, this sets whether it is hanging from the block above it.
+        // For a tripwire, this sets whether a tripwire hook or string forms a complete tripwire circuit and is ready to trigger.
+        // Updating the property on a tripwire hook will change the texture to indicate a connected string, but will not have
+        // any effect when used on the tripwire string itself. It may however still be used to check whether the string forms a circuit.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttached.java]: These lines are split really awkwardly. If you must insert a newline, place it after a period or a comma, not just in the middle.

**Pull request merged**